test/android/core/javatests/src/org/xwalk/core/xwview/test/SettingsTest.java

```diff
@@ -17,10 +19,80 @@
     private static final String LANGUAGE = "jp";
+    private TestWebServer mWebServer;
 
     @Override
     public void setUp() throws Exception {
         super.setUp();
+        mWebServer = TestWebServer.start();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        mWebServer.shutdown();
+        super.tearDown();
+    }
+
+    @MediumTest
+    @Feature({"Settings"})
+    public void testCacheMode() throws Throwable {
+        final String htmlPath = "/testCacheMode.html";
+        final String url = mWebServer.setResponse(htmlPath, "response", null);
+        final String htmlNotInCachePath = "/testCacheMode-not-in-cache.html";
+        final String urlNotInCache = mWebServer.setResponse(htmlNotInCachePath, "", null);
+
+        clearCacheOnUiThread(true);
+        assertEquals(XWalkSettings.LOAD_DEFAULT, getCacheMode());
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ELSE_NETWORK);
+        loadUrlSync(url);
+        assertEquals(1, mWebServer.getRequestCount(htmlPath));
+        loadUrlSync(url);
+        assertEquals(1, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_NO_CACHE);
+        loadUrlSync(url);
+        assertEquals(2, mWebServer.getRequestCount(htmlPath));
+        loadUrlSync(url);
+        assertEquals(3, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ONLY);
+        loadUrlSync(url);
+        assertEquals(3, mWebServer.getRequestCount(htmlPath));
+        loadUrlSync(url);
+        assertEquals(3, mWebServer.getRequestCount(htmlPath));
+
+        loadUrlSyncAndExpectError(urlNotInCache);
+        assertEquals(0, mWebServer.getRequestCount(htmlNotInCachePath));
+    }
+
+    @MediumTest
+    @Feature({"Settings"})
+    // As our implementation of network loads blocking uses the same net::URLRequest settings, make
+    // sure that setting cache mode doesn't accidentally enable network loads.  The reference
+    // behaviour is that when network loads are blocked, setting cache mode has no effect.
+    public void testCacheModeWithBlockedNetworkLoads() throws Throwable {
+        final String htmlPath = "/testCacheModeWithBlockedNetworkLoads.html";
+        final String url = mWebServer.setResponse(htmlPath, "response", null);
+
+        clearCacheOnUiThread(true);
+        assertEquals(XWalkSettings.LOAD_DEFAULT, getCacheMode());
+
+        setBlockNetworkLoads(true);
+        loadUrlSyncAndExpectError(url);
+        assertEquals(0, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ELSE_NETWORK);
+        loadUrlSyncAndExpectError(url);
+        assertEquals(0, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_NO_CACHE);
+        loadUrlSyncAndExpectError(url);
+        assertEquals(0, mWebServer.getRequestCount(htmlPath));
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ONLY);
```

Comment:
The function of `clearCacheOnUiThread` cleared the cache, do the test case of  LOAD_CACHE_ONLY need  which don't load resources from network?